python/lib/evaluations/dmod/evaluations/threshold/disk.py

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env python3
+import io
+import typing
+import os
+import re
+import inspect
+
+import pandas
+import jsonpath_ng as jsonpath
+
+from jsonpath_ng.ext import parse as create_expression
+
+from .. import specification
+from .. import jsonquery
+from .. import util
+
+from . import retriever
+
+
+def get_datasource(datasource_definition: specification.DataSourceSpecification) -> retriever.ThresholdRetriever:
+    return __FORMAT_MAPPING[datasource_definition.backend.format](datasource_definition)
+
+
+class JSONThresholdRetriever(retriever.ThresholdRetriever):
```

Comment:
While more clear in the abstractions (see comment below), it feels like there is a lot of overlap in the _JSONDataRetriever, _JSONThresholdRetriever_, and _JSONRetriever_ (for crosswalk) implementations.  It would be nice to abstract some parts; e.g., perhaps to a mixin or something.   But this isn't necessary and could certainly be deferred until later.

Reply:
I'm probably going to rename `JSONRetriever` to `JSONCrosswalkRetriever`, but I think most if not all of the overlap will be resolved via the distillation of `ThresholdRetriever` as a root class. Those classes are pretty different outside those common functions.